code/modules/vehicles/multitile/tankvendor.dm

```diff
@@ -0,0 +1,290 @@
+					dat += "<b>Current rounds</b>: [A] [A.current_rounds]/[A.max_rounds]<br>"
+					dat += "<b>Caliber</b>: [A.caliber]<br>"
+					var/obj/item/hardpoint/H = A.gun_type
+					dat += "<b>Supported Hardpoint</b>: [initial(H.name)]<br>"
+
+		if(TANKFAB_PRINTER)
+			dat += "<h3>Armor:</h3>"
+			for(var/build_type in subtypesof(/obj/item/hardpoint/armor))
+				var/obj/item/hardpoint/armor/AR = build_type
+				if(!initial(AR.buyable))
+					continue
+				var/build_name = initial(AR.name)
+				var/build_cost = initial(AR.point_cost)
+				if(build_cost)
+					dat += "<a href='byond://?src=\ref[src];produce=[build_type];cost=[build_cost]'>[build_name] ([build_cost])</a><br>"
+
+			dat += "<h3>Primary Weapon:</h3>"
+			for(var/build_type in subtypesof(/obj/item/hardpoint/primary))
+				var/obj/item/hardpoint/primary/PR = build_type
+				if(!initial(PR.buyable))
+					continue
+				var/build_name = initial(PR.name)
+				var/build_cost = initial(PR.point_cost)
+				if(build_cost)
+					dat += "<a href='byond://?src=\ref[src];produce=[build_type];cost=[build_cost]'>[build_name] ([build_cost])</a><br>"
+
+			dat += "<h3>Secondary Weapon:</h3>"
+			for(var/build_type in subtypesof(/obj/item/hardpoint/secondary))
+				var/obj/item/hardpoint/secondary/SE = build_type
+				if(!initial(SE.buyable))
+					continue
+				var/build_name = initial(SE.name)
+				var/build_cost = initial(SE.point_cost)
+				if(build_cost)
+					dat += "<a href='byond://?src=\ref[src];produce=[build_type];cost=[build_cost]'>[build_name] ([build_cost])</a><br>"
+
+			dat += "<h3>Support Module:</h3>"
+			for(var/build_type in subtypesof(/obj/item/hardpoint/support))
+				var/obj/item/hardpoint/support/SP = build_type
+				if(!initial(SP.buyable))
+					continue
+				var/build_name = initial(SP.name)
+				var/build_cost = initial(SP.point_cost)
+				if(build_cost)
+					dat += "<a href='byond://?src=\ref[src];produce=[build_type];cost=[build_cost]'>[build_name] ([build_cost])</a><br>"
+
+			dat += "<h3>Treads:</h3>"
+			for(var/build_type in subtypesof(/obj/item/hardpoint/treads))
+				var/obj/item/hardpoint/treads/TR = build_type
+				if(!initial(TR.buyable))
+					continue
+				var/build_name = initial(TR.name)
+				var/build_cost = initial(TR.point_cost)
+				if(build_cost)
+					dat += "<a href='byond://?src=\ref[src];produce=[build_type];cost=[build_cost]'>[build_name] ([build_cost])</a><br>"
+
+			dat += "<h3>Weapon Ammo:</h3>"
+			for(var/build_type in subtypesof(/obj/item/ammo_magazine/tank))
+				var/obj/item/ammo_magazine/tank/AM = build_type
+				var/build_name = initial(AM.name)
+				var/build_cost = initial(AM.point_cost)
+				if(build_cost)
+					dat += "<a href='byond://?src=\ref[src];produce=[build_type];cost=[build_cost]'>[build_name] ([build_cost])</a><br>"
+
+	var/datum/browser/popup = new(user, "dropship_part_fab", "<div align='center'>Tank Part Fabricator</div>")
+	popup.set_content(dat)
+	popup.open(FALSE)
+	onclose(user, "dropship_part_fab")
```

Comment:
The argument to open() is if you want to setup onclose.

Reply:
care to explain a bit more?

Reply:
do open(TRUE) so you don't have to setup onclose.

Reply:
Thanks.